rs/boundary_node/rate_limits/canister/access_control.rs

```diff
@@ -0,0 +1,54 @@
+use candid::Principal;
+use mockall::automock;
+
+use crate::storage::API_BOUNDARY_NODE_PRINCIPALS;
+
+const FULL_ACCESS_ID: &str = "imx2d-dctwe-ircfz-emzus-bihdn-aoyzy-lkkdi-vi5vw-npnik-noxiy-mae";
+const FULL_READ_TESTING_ID: &str = "un4fu-tqaaa-aaaab-qadjq-cai"; // TODO: remove this
+
+#[automock]
+pub trait ResolveAccessLevel {
+    fn get_access_level(&self) -> AccessLevel;
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum AccessLevelError {}
+
+#[derive(PartialEq, Eq)]
+pub enum AccessLevel {
+    FullAccess,
+    FullRead,
+    RestrictedRead,
```

Comment:
Maybe we should call it something like `Default` or `Unauthorized`. Now, it sounds like this is still a special permission and there is something like `NoRead`.

Reply:
`Unauthorized` could be a valid variant only for `add_config()/disclose()` operations, however `get_config()/get_rule_bu_id()` operations are always "authorized"  just the access rights can be different. `Default` sounds a bit vague to me, a comment could resolve it though. Let's discuss it more. I introduced this enum based on the access level to the stored data, it was meant to be agnostic to specific canister methods.